### PR TITLE
Import error handling

### PR DESCRIPTION
**Pull request by SergeyGaluzo**
Removes incorrect usage of retriable exception.
Replaces "global" retries by local.
Sets RetriableException as obsolete.
Adds e2e tests for not retriable SQL exceptions.

Fixes
https://microsofthealth.visualstudio.com/Health/_workitems/edit/103819/

### CODE REVIEW

File: src/Microsoft.Health.Fhir.Core/Features/Operations/Import/ImportErrorStore.cs [hunk 1]
```diff
@@ -59,17 +60,24 @@ public async Task UploadErrorsAsync(string[] importErrors, CancellationToken can
                     await writer.WriteLineAsync(error);
                 }
 
+#pragma warning disable CA2016 // .NET 6 does not have cancel. Remove when we switch to .NET8
```
Comment (brendankowitz) [inline src/Microsoft.Health.Fhir.Core/Features/Operations/Import/ImportErrorStore.cs:63]: I'd prefer the method level Suppression over this

File: src/Microsoft.Health.Fhir.Core/Features/Operations/Import/ImportErrorStore.cs [hunk 2]
```diff
@@ -59,17 +60,24 @@ public async Task UploadErrorsAsync(string[] importErrors, CancellationToken can
...
             {
                 _logger.LogWarning(ex, "Failed to upload import error log.");
-                throw new RetriableJobException(ex.Message, ex);
+                if (retries++ < 3)
+                {
+                    await Task.Delay(5000, cancellationToken);
+                    goto retry;
+                }
+
+                throw;
```
Comment (brendankowitz) [inline src/Microsoft.Health.Fhir.Core/Features/Operations/Import/ImportErrorStore.cs:80]: We should use Polly for retries going forward.

File: src/Microsoft.Health.Fhir.SqlServer/Features/Operations/Import/ImportOrchestratorJob.cs [hunk 3]
```diff
@@ -412,12 +383,24 @@ private async Task<IList<long>> EnqueueProcessingJobsAsync(IEnumerable<InputReso
...
+                var error = new ImportJobErrorResult() { ErrorMessage = ex.Message, ErrorDetails = ex.ToString(), HttpStatusCode = HttpStatusCode.BadRequest };
+                throw new JobExecutionException(message, error, ex);
+            }
+            catch (SqlException ex) when (ex.IsExecutionTimeout())
+            {
+                _logger.LogJobWarning(ex, orchestratorInfo, ex.Message);
+                if (retries++ < 3)
+                {
+                    await Task.Delay(5000, cancellationToken);
+                    goto retry;
```
[the remark this answers is not in the document]
Comment (SergeyGaluzo) [reply src/Microsoft.Health.Fhir.SqlServer/Features/Operations/Import/ImportOrchestratorJob.cs:395]: Fixed

File: src/Microsoft.Health.Fhir.Core/Features/Operations/Import/ImportErrorStore.cs (hunk 2 above)
Comment (SergeyGaluzo) [reply src/Microsoft.Health.Fhir.Core/Features/Operations/Import/ImportErrorStore.cs:80]: Will fix

File: src/Microsoft.Health.Fhir.Core/Features/Operations/Import/ImportErrorStore.cs (hunk 1 above)
Comment (SergeyGaluzo) [reply src/Microsoft.Health.Fhir.Core/Features/Operations/Import/ImportErrorStore.cs:63]: Removed

File: src/Microsoft.Health.TaskManagement/RetriableJobException.cs [hunk 4]
```diff
@@ -8,6 +8,7 @@
 
 namespace Microsoft.Health.JobManagement
 {
+    [Obsolete("RetriableJobException class is deprecated. Replace by in-place retries.")]
```
[the remark this answers is not in the document]
Comment (SergeyGaluzo) [reply src/Microsoft.Health.TaskManagement/RetriableJobException.cs:11]: It is not used. I just wanted to keep it for a while, just in case.

**Review by LTA-Thinking — APPROVED** [review]

**Pull request merged**